src/test/K6/src/tests/app/performance/efm.js

```diff
@@ -0,0 +1,115 @@
+import * as setUpData from '../../../setup.js';
+
+const appOwner = "ttd";
+const level2App = "eformidling-app";
+const environment = "tt02";
+const fileName = 'users_' + environment + '.json';
+const toDelete = __ENV.delete ? __ENV.delete.toLowerCase() : 'true';
+const hardDelete = __ENV.harddelete ? __ENV.harddelete.toLowerCase() : 'true';
+
+let pdfAttachment = open('../../../data/test_file_pdf.pdf', 'b');
+let instanceFormDataXml = open('../../../data/' + level2App + '.xml');
+let users = new SharedArray('test users', function () {
+  var usersArray = JSON.parse(open('../../../data/' + fileName));
+  return usersArray;
+});
+const usersCount = users.length;
+
+export const options = {
+  thresholds: {
+    errors: ['count<1'],
+  },
+};
+
+//Tests for App API: eFormidling-app
+export default function () {
+  var userNumber = (__VU - 1) % usersCount;
+  var instanceId, dataId, res, success;
+  const attachmentDataType ="9dd34a1e-e9f5-4f64-ab47-3a1a4a362e92";
+
+  try {
+    var userSSN = users[userNumber].username;
+    var userPwd = users[userNumber].password;
+  } catch (error) {
+    stopIterationOnFail('Testdata missing', false, null);
+  }
+
+  var aspxauthCookie = setUpData.authenticateUser(userSSN, userPwd);
+  const runtimeToken = setUpData.getAltinnStudioRuntimeToken(aspxauthCookie);
+  setUpData.clearCookies();
+  const partyId = users[userNumber].partyid;
+
+  //Test to create an instance with App api and validate the response
+  res = appInstances.postInstance(runtimeToken, partyId, appOwner, level2App);
+  success = check(res, {
+    'E2E App POST Create Instance status is 201': (r) => r.status === 201,
+  });
+  addErrorCount(success);
+  stopIterationOnFail('E2E App POST Create Instance', success, res);
+
+  try {
+    dataId = appData.findDataId(res.body);
+    instanceId = platformInstances.findInstanceId(res.body);
+  } catch (error) {
+    stopIterationOnFail('Instance id and data id not retrieved', false, null);
+  }
+
+  //Test to edit a form data in an instance with App APi and validate the response
+  res = appData.putDataById(runtimeToken, partyId, instanceId, dataId, null, instanceFormDataXml, appOwner, level2App);
+  success = check(res, {
+    'E2E PUT Edit Data by Id status is 201': (r) => r.status === 201,
+  });
+  addErrorCount(success);
+  stopIterationOnFail('E2E PUT Edit Data by Id', success, res);
+
+
+  //upload a valid attachment to an instance with App API
+  res = appData.postData(runtimeToken, partyId, instanceId, attachmentDataType, pdfAttachment, 'pdf', appOwner, level2App);
+
+
+    //Test to get next process of an app instance again and verify response code  to be 200
+    res = appProcess.putCompleteProcess(runtimeToken, partyId, instanceId, appOwner, level2App);
+    success = check(res, {
+      'E2E App GET Next process element id': (r) => r.status === 200,
+    });
+    addErrorCount(success);
+    stopIterationOnFail('Unable to get next element id', success, res);
+
```

Comment:
remember to format the files with [scripts](https://github.com/Altinn/altinn-studio/tree/master/src/test/K6#format-files-with-prettier)